### PR TITLE
Add rocks.syng.Syng

### PR DESCRIPTION
**Pull request by christofsteel**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

Syng is a karaoke software, consisting of a backend server, a web frontend and a playback client. This is the playback client, that is used to host karaoke events. Configuration is done via a QT GUI, afterwards the client connects to a backend server and waits for songs to be queued.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am the author of the project.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: mpv.yaml
Comment (hfiguiere) [inline mpv.yaml:1]: since it's not autogenerated, this should be inline in the main manifest.

File: rocks.syng.Syng.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+sdk: org.kde.Sdk
+base: "com.riverbankcomputing.PyQt.BaseApp"
+base-version: "6.7"
+cleanup-commands:
+  - /app/cleanup-BaseApp.sh
+build-options:
+  env:
+    - BASEAPP_REMOVE_WEBENGINE=1
+finish-args:
+  - --env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess - /app/cleanup-BaseApp.sh
```
Comment (hfiguiere) [inline rocks.syng.Syng.yaml:13]:
```suggestion
  - --env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess
```

File: rocks.syng.Syng.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  # X11 + XShm access
+  - --socket=fallback-x11
+  - --share=ipc
+  - --socket=wayland
+  # Acceleration
+  - --device=dri
+  # Sound
+  - --socket=pulseaudio
+  # Ability to edit/save files anywhere
+  - --filesystem=host
```
Comment (hfiguiere) [inline rocks.syng.Syng.yaml:23]: anywhere? Really?

**Comment by hfiguiere** [comment]
bot, build rocks.syng.Syng

File: rocks.syng.Syng.yaml (hunk 2 above)
Comment (christofsteel) [reply rocks.syng.Syng.yaml:23]: comment is misleading (will fix), this needs to be set for playback of karaoke files, that could be stored anywhere on the system, will be changed to `--filesystem=host:ro`

File: mpv.yaml
Comment (christofsteel) [reply mpv.yaml:1]: done

**Comment by christofsteel** [comment]
(old) Build went through, and I incorporated your comments. Anything else, that should be done?

**Comment by hfiguiere** [comment]
bot, build rocks.syng.Syng

File: python3-poetry-core.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Generated with flatpak-pip-generator --yaml poetry-core==1.9.0
+name: python3-poetry-core
+buildsystem: simple
+build-commands:
```
Comment (bbhtt) [inline python3-poetry-core.yaml:4]: this is a build-only dependency ? If so it should be cleaned up using cleanup with `- "*"`

Comment (christofsteel) [reply python3-poetry-core.yaml:4]: yes, and done

**Comment by christofsteel** [comment]
Am I allowed (and should I) initiate the buildbot?

**Comment by bbhtt** [comment]
sure

bot, build rocks.syng.Syng

**Comment by christofsteel** [comment]
bot, build rocks.syng.Syng

**Comment by christofsteel** [comment]
There seemed to be issues with the svg icon (renders in inkscape, but nowhere else) so I removed it. Additionally, I trimmed the dependencies of mpv, so the resulting flatpak is much smaller and builds faster.

**Comment by christofsteel** [comment]
Build runs through, package is smaller, anything further?

File: rocks.syng.Syng.yaml [hunk 4]
```diff
@@ -0,0 +1,406 @@
+id: rocks.syng.Syng
```
[the remark this answers is not in the document]
Comment (christofsteel) [reply rocks.syng.Syng.yaml:1]: If you enter an unused room code (like "flathubtest"), you will get an instruction, with a link to the repository, if that is enough for you.

[the remark this answers is not in the document]
Comment (christofsteel) [reply rocks.syng.Syng.yaml:1]: And you need to click on connect afterwards

[the remark this answers is not in the document]
Comment (bbhtt) [reply rocks.syng.Syng.yaml:1]: > If you enter an unused room code (like "flathubtest"), you will get an instruction, with a link to the repository, if that is enough for you.


shows no such room if I click connect

image

[the remark this answers is not in the document]
Comment (christofsteel) [reply rocks.syng.Syng.yaml:1]: No, what I meant is, that there now is a link (the blue Syng text), that links to the repository.

[the remark this answers is not in the document]
Comment (bbhtt) [reply rocks.syng.Syng.yaml:1]: Ah ok thanks.

**Pull request closed without merging**